src/database/entities/credential-provider.entity.ts

```diff
@@ -0,0 +1,120 @@
+import { BeforeInsert, BeforeUpdate, Column, Entity, OneToMany } from 'typeorm';
+import { ProviderConfigurationEntity } from './provider-configuration.entity.js';
+import * as dotenv from 'dotenv';
+
+dotenv.config();
+
+@Entity('credentialProvider')
+export class CredentialProviderEntity {
+	@Column({
+		type: 'varchar',
+		primary: true,
+	})
+	providerId!: string;
+
+	@Column({
+		type: 'varchar',
+		nullable: false,
+	})
+	name!: string;
+
+	@Column({
+		type: 'text',
+		nullable: true,
+	})
+	description?: string;
+
+	@Column({
+		type: 'varchar',
+		nullable: false,
+	})
+	providerType!: string; // 'credential-all', 'credential-issuance', 'credential-verification', 'etc'
+
+	@Column({
+		type: 'json',
+		nullable: false,
+		default: '[]',
+	})
+	supportedFormats!: string[];
+
+	@Column({
+		type: 'json',
+		nullable: false,
+		default: '[]',
+	})
+	supportedProtocols!: string[];
+
+	@Column({
+		type: 'json',
+		nullable: false,
+		default: '[]',
+	})
+	capabilities!: string[];
+
+	@Column({
+		type: 'text',
+		nullable: true,
+	})
+	logoUrl?: string;
+
+	@Column({
```

Comment:
We can skip the fields and retain only 

- providerId
- name
- createdAt
- updatedAt 
- deprecated